pkg/resources/oauth_integration_for_custom_clients.go

```diff
@@ -0,0 +1,666 @@
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
+
+		integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		if c := integration.Category; c != sdk.SecurityIntegrationCategory {
+			return diag.FromErr(fmt.Errorf("expected %v to be a %s integration, got %v", id, sdk.SecurityIntegrationCategory, c))
+		}
+
+		if err := d.Set("name", integration.Name); err != nil {
+			return diag.FromErr(err)
+		}
+
+		if err := d.Set("comment", integration.Comment); err != nil {
+			return diag.FromErr(err)
+		}
+
+		oauthClientType, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "OAUTH_CLIENT_TYPE"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find oauth client type, err = %w", err))
+		}
+		if err := d.Set("oauth_client_type", oauthClientType.Value); err != nil {
+			return diag.FromErr(err)
+		}
+
+		oauthRedirectUri, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "OAUTH_REDIRECT_URI"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find oauth redirect uri, err = %w", err))
+		}
+		if err := d.Set("oauth_redirect_uri", oauthRedirectUri.Value); err != nil {
+			return diag.FromErr(err)
+		}
+
+		preAuthorizedRolesList, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "PRE_AUTHORIZED_ROLES_LIST"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find pre authorized roles list, err = %w", err))
+		}
+		var preAuthorizedRoles []string
+		if len(preAuthorizedRolesList.Value) > 0 {
+			preAuthorizedRoles = strings.Split(preAuthorizedRolesList.Value, ",")
+		}
+		if err := d.Set("pre_authorized_roles_list", preAuthorizedRoles); err != nil {
+			return diag.FromErr(err)
+		}
+
+		blockedRolesList, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "BLOCKED_ROLES_LIST"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find pre authorized roles list, err = %w", err))
+		}
+		var blockedRoles []string
+		if len(blockedRolesList.Value) > 0 {
+			blockedRoles = strings.Split(blockedRolesList.Value, ",")
+		}
+		if err := d.Set("blocked_roles_list", blockedRoles); err != nil {
+			return diag.FromErr(err)
+		}
+
+		networkPolicy, err := collections.FindOne(integrationProperties, func(property sdk.SecurityIntegrationProperty) bool {
+			return property.Name == "NETWORK_POLICY"
+		})
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("failed to find network policy, err = %w", err))
+		}
+		if err := d.Set("network_policy", sdk.NewAccountObjectIdentifier(networkPolicy.Value).Name()); err != nil {
```

Comment:
not FullyQualifiedName?

Reply:
It's account object, so it doesn't matter that much, in warehouse resource monitor was done this way, so I did the same here.